main.go

```diff
@@ -117,18 +125,45 @@ func main() { //nolint:funlen
 
 	ctrl.SetLogger(logger.ConfigLoggers(logmode))
 
+	cacheOptions := cache.Options{
+		Scheme: scheme,
+		ByObject: map[client.Object]cache.ByObject{
+			// all CRD because it is mainly for pipeline v1 and v2
+			&apiextv1.CustomResourceDefinition{}: {},
```

Comment:
Did you consider trimming the cache by removing parts of the object we do not care about - we can use [TransformFunc](https://github.com/kubernetes-sigs/controller-runtime/blob/v0.16.3/pkg/cache/cache.go#L250-L255)  for that. Perhaps a follow-up.

Reply:
we have a similar case https://issues.redhat.com/browse/RHOAIENG-6875, i will add these in the ticket to follow up.